### PR TITLE
update coding style

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**

**Changes:**
1. merge/add/change naming rules
2. add index
3. fix some error (e.g., class and its begin brace are not separate)

### CODE REVIEW

File: guides/coding_style/c++/style.md [hunk 1]
```diff
@@ -122,58 +141,109 @@ class ListHelp {
...
+```C++
+const int Name;
+constexpr int VariableName;
+
+enum class ExampleEnum
+{
+    One,
+    TwoTwo
+};
+```
```
Comment (iattempt) [inline guides/coding_style/c++/style.md]: now, the constant variables (const, constexpr, enumerator) using same rules

File: guides/coding_style/c++/style.md [hunk 2]
```diff
@@ -122,58 +141,109 @@ class ListHelp {
...
-**Valid:**
+### Exceptions to Naming Rules
+If you are naming something that is analogous to an existing C++ entity then you can follow the existing naming convention scheme.
 ```C++
-const int name;
-constexpr int constantName;
+sparse_hash_set<> ...  // STL-like entity; follows STL naming conventions
+long BUFFER_MAX ...    // a constant, as in INT_MAX
+using diff_t ...       // as in ptrdiff_t
 ```
```
Comment (iattempt) [inline guides/coding_style/c++/style.md]: add exceptions for conventions

Comment (iattempt) [inline guides/coding_style/c++/style.md]: add exceptions for conventions

File: guides/coding_style/c++/style.md [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
+        <li><a href="#macro-names">Macro Names</a></li>
+        <li><a href="#exceptions-to-naming-rules">Exceptions to Naming Rules</a></li>
+    </ul>
+  </details>
+- [Braces](#braces)
+- [Functions](#functions)
+- [Classes](#classes)
+- [Enumerations](#enumerations)
+- [Alias](#alias)
+
```
Comment (iattempt) [inline guides/coding_style/c++/style.md]: add index

**Comment by iattempt** [comment]
fixed conflicts

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**